opennebula/resource_opennebula_virtual_machine.go

```diff
@@ -828,7 +828,7 @@ func resourceOpennebulaVirtualMachineUpdate(d *schema.ResourceData, meta interfa
 		// get unique elements of each list of configs
 		toDetach, toAttach := diffListConfig(newDisksCfg, attachedDisksCfg,
 			&schema.Resource{
-				Schema: diskVMFields(),
+				Schema: diskFields(),
```

Comment:
I don't get this one ? 
Where did you rename the function ?

Reply:
I didn't renamed the function, I just passed less attributes to the diff helper in removing `computed` only attributes:
```
func diskVMFields() map[string]*schema.Schema {
	return mergeSchemas(diskFields(), diskComputedVMFields())
}
```
Here we only want to react on user changes.

Anyway, If I understand right, this should have no impact: inside of `diffListConfig` we have some `Set` data structures built like this:
```
set := schema.NewSet(schema.HashResource(s), []interface{}{})
```
Here `s`  contains `&schema.Resource{ Schema: diskVMFields() } `).

If we dig a bit to see what `HashResource` does ( [see source](https://github.com/hashicorp/terraform/blob/v0.12.12/helper/schema/set.go#L29) ), it compute a hash from bytes returned by `SerializeResourceForHash`.

And inside of  `SerializeResourceForHash` there is this piece of code that iterate on schema attributes ( [see source](https://github.com/hashicorp/terraform/blob/v0.12.12/helper/schema/serialize.go#L87) ):
```
...
	for _, k := range keys {
		innerSchema := sm[k]
		// Skip attributes that are not user-provided. Computed attributes
		// do not contribute to the hash since their ultimate value cannot
		// be known at plan/diff time.
		if !(innerSchema.Required || innerSchema.Optional) {
			continue
		}
...
```

Reply:
Ok thanks